app/controllers/likes_controller.rb

```diff
@@ -7,8 +7,7 @@ def create
   end
 
   def destroy
-    like = current_user.likes.find_by(id:params[:id])
-    like.delete
+    current_user.likes.find_by(id:params[:id]).delete
```

Comment:
意味合いが違います、、、
1行にするなら下記のいずれかです

```suggestion
    current_user.likes.find_by(post_id: @post.id).destroy
```

```suggestion
    Like.find_by(id: params[:id]).destroy
```

あとdeleteよりdestroyを使う方が無難です。
違いはググってみてください。